subsys/nrf_security/src/drivers/cracen/cracenpsa/src/sign.c

```diff
@@ -306,14 +340,30 @@ static psa_status_t cracen_signature_ecc_sign(int message, const psa_key_attribu
 	sign.r = (char *)signature;
 	sign.s = (char *)signature + *signature_length / 2;
 
-	if (message) {
+	/* ED25519PH requires prehashing and supports sign and verify message
+	 * the message is therefore hashed here before si_sig_verify is called
+	 */
+	if (alg == PSA_ALG_ED25519PH && message) {
+		uint8_t status;
+		uint8_t hash[64];
+		size_t output_len;
+
+		status = psa_hash_compute(PSA_ALG_SHA_512, input,
+			input_length, hash, sizeof(hash), &output_len);
+
 		si_sig_create_sign(&t, &privkey, &sign);
-		si_task_consume(&t, (char *)input, input_length);
+		si_task_consume(&t, (char *)hash, sizeof(hash));
 	} else {
-		si_sig_create_sign_digest(&t, &privkey, &sign);
-		si_task_consume(&t, (char *)input, sx_hash_get_alg_digestsz(privkey.hashalg));
-	}
+		if (message) {
+			si_sig_create_sign(&t, &privkey, &sign);
+		} else {
+			si_sig_create_sign_digest(&t, &privkey, &sign);
```

Comment:
Oh I hadn't noticed that those two functions weren't the same... :thinking:

Reply:
Aaah. They technically  are the same in the case of ED25519PH, but in other cases it might not be